recipes/mbedtls/all/conanfile.py

```diff
@@ -132,6 +142,9 @@ def package_info(self):
             for component in self.cpp_info.components:
                 self.cpp_info.components[component].requires.append("zlib::zlib")
 
+        if self.options.enable_threading:
+            self.cpp_info.defines.extend(["MBEDTLS_THREADING_C=1", "MBEDTLS_THREADING_PTHREAD=1"])
```

Comment:
```suggestion
            self.cpp_info.components["mbedcrypto"].defines.extend(["MBEDTLS_THREADING_C=1", "MBEDTLS_THREADING_PTHREAD=1"])
```

Changing the defines to be on the mbedcrypto component solved the issue I was seeing https://github.com/conan-io/conan-center-index/pull/23366#issuecomment-2039222586

With this only on `self.cpp_info.defines` I'm not seeing it added to mbedtls_DEFINITIONS_DEBUG in the generated generators/MbedTLS-debug-x86_64-data.cmake